### ISSUE TITLE
Plugin should remove LSP related code and rely on nvim-lspconfig

### ISSUE THREAD

**Issue by eshepelyuk** [kcl-lang/vim-kcl#11]
Hello,

Currently plugin relies on combination of plugin manager setting (like lazyvim) and internal code to manage start of LSP servers.
This causes few bugs (like in #10 ) and removes portability between different plugin managers.

I've been playing with my fork and `nvim-lspconfig` and I realize that lsp related code can be completely removed 
and offloaded to `nvim-lspconfig`.

Having `nvim-lspconfig` plugin added via your plugin manager and configuring `nvim-lspconfig` like this 

```lua
local server_config = require("lspconfig.configs")
local util = require("lspconfig.util")

server_config.kcl = {
	default_config = {},
}

require("lspconfig").kcl.setup({
	cmd = { "kcl-language-server" },
	filetypes = { "kcl" },
	root_dir = util.root_pattern(".git"),
})
```

It's possible to remove `lsp/init.lua` and `lsp/handlers.lua` and everything will keep working.
More over, those files are overriding user's LSP settings, like diagnostics, that is undesirable.

So basically, this plugin may only provide syntax highligh and comment string - and with `nvim-lspconfig` it will be enough.

**Comment by Peefy** [comment]
cc @He1pa

**Comment by He1pa** [comment]
Thanks for your suggestion, I'll test it

**Comment by eshepelyuk** [comment]
> Thanks for your suggestion, I'll test it

* This is my fork with removed LSP https://github.com/eshepelyuk/kcl.nvim
* This is minimal config using `vim-plug`, `test.init.vim`
```vim 
    call plug#begin(stdpath('data') . '/plugged')

    Plug 'eshepelyuk/kcl.nvim'
    Plug 'neovim/nvim-lspconfig'

    call plug#end()

    lua << EOF
    local server_config = require("lspconfig.configs")
    local util = require("lspconfig.util")

    server_config.kcl = {
        default_config = {},
    }

    require("lspconfig").kcl.setup({
        cmd = { "kcl-language-server" },
        filetypes = { "kcl" },
        root_dir = util.root_pattern(".git"),
    })
    EOF
````
* Run NeoVim with `nvim -u test.init.vim some.k`

**Comment by eshepelyuk** [comment]
Here's a screenshot running `: LspInfo` from `nvin-lspconfig`
![image](https://github.com/user-attachments/assets/de2ece2f-4e53-45d9-8e2c-4bf0e31a8df9)

**Comment by He1pa** [comment]
Thanks a lot, I tested it and it works for both lazy.vim and vim-plug. I have submitted a PR to modify them

**Comment by eshepelyuk** [comment]
@He1pa all works fine, thankx a lot.
